jobbergateapi2/apps/job_scripts/routers.py

```diff
@@ -219,5 +222,92 @@ async def job_script_list(all: Optional[bool] = Query(None), current_user: User
     return job_scripts
 
 
+@router.delete(
+    "/job-scripts/{job_script_id}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    description="Endpoint to delete job script",
+)
+async def job_script_delete(
+    current_user: User = Depends(get_current_user),
+    job_script_id: int = Query(..., description="id of the application to delete"),
+):
+    """
+    Delete job_script given its id.
+    """
+    where_stmt = (job_scripts_table.c.id == job_script_id) & (
+        job_scripts_table.c.job_script_owner_id == current_user.id
+    )
+
+    get_query = job_scripts_table.select().where(where_stmt)
+    raw_job_script = await database.fetch_one(get_query)
+    if not raw_job_script:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"JobScript with id={job_script_id} not found for user={current_user.id}",
+        )
+
+    delete_query = job_scripts_table.delete().where(where_stmt)
+    await database.execute(delete_query)
+
+
+@router.put(
+    "/job-scripts/{job_script_id}",
+    status_code=status.HTTP_201_CREATED,
+    description="Endpoint to update a job_script given the id",
+    response_model=JobScript,
+)
+async def job_script_update(
+    job_script_id: int = Query(...),
+    job_script_name: Optional[str] = Form(None),
+    job_script_description: Optional[str] = Form(None),
+    job_script_data_as_string: Optional[str] = Form(None),
+    current_user: User = Depends(get_current_user),
+):
+    """
+    Update a job_script given its id.
+    """
+    query = job_scripts_table.select().where(
+        (job_scripts_table.c.id == job_script_id)
+        & (job_scripts_table.c.job_script_owner_id == current_user.id)
+    )
+    raw_job_script = await database.fetch_one(query)
+
+    if not raw_job_script:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"JobScript with id={job_script_id} not found for user={current_user.id}",
+        )
+    job_script = JobScript.parse_obj(raw_job_script)
+
+    job_script_data = JobScript(**job_script.dict())
+    if job_script_name is not None:
+        job_script_data.job_script_name = job_script_name
+    if job_script_description is not None:
+        job_script_data.job_script_description = job_script_description
+    if job_script_data_as_string is not None:
+        job_script_data.is_superuser = job_script_data_as_string
+
+    job_script_data.updated_at = datetime.utcnow()
+
+    values = {
+        "job_script_name": job_script_data.job_script_name,
+        "job_script_description": job_script_data.job_script_description,
+        "job_script_data_as_string": job_script_data.job_script_data_as_string,
+        "updated_at": job_script_data.updated_at,
+    }
+    validated_values = {key: value for key, value in values.items() if value is not None}
```

Comment:
Why do we need to use a separate values dict? I understand we load our data ( **job_script.dict() ) and then need to compile a dict of all of these key-value pairs that are not None. I believe we could do this in fewer lines without using three dicts, job_script_data, values, and validated values. Something along the lines of:
return {key: value for key, value in JobScript(**job_script.dict()) if value is not None}
Let me know if I am missing something here.

Reply:
I don't get your suggestion, but we need to make sure the key don't exists in the dict we use for the `values` in the `q_update`, to ensure updating only the changed fields (the ones in the request body). Can you develop more you suggestion?
And we always should prioritize readability instead of fewer lines, and I tried to be as most readable as possible in this function.